### PR TITLE
Disallowing out of date range

### LINKED ISSUE #9030: Disallow picking out of range month or year from month/year selector in date input dropdown
**Describe the bug**
Using the month selector, I can go to any month in the calendar (even ones beyond the allowed range). However, once a month out of the range is selected, I am not able use the navigation button to move towards the month that is in range.

**Expected behavior**
- [ ] User should not be able to pick a month or year that is out of range in the first place from the month/year selector view.

**Screenshots**

Image

Image

### PR DESCRIPTION
**Pull request by Rishith25**
## Proposed Changes

- Fixes #9030 
- Restrict month/year selection to only valid months and years, preventing out-of-range selections.
- Ensure that the navigation buttons are always functional, even when an invalid month/year has been selected.
- Add validation to ensure that invalid months or years cannot be selected via the month/year selector.

@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

## Summary by CodeRabbit

- **New Features**
	- Enhanced date selection component with improved logic for selecting dates, months, and years.
	- New validation methods ensure selections are within defined minimum and maximum limits.
	- Conditional disabling of selection buttons based on constraints.
	- Improved error handling provides user feedback for out-of-range selections.

### CODE REVIEW

**Comment by Rishith25** [comment]
@rithviknishad @nihal467 can you please review it.

File: src/components/Common/DateInputV2.tsx [hunk 1]
```diff
@@ -108,11 +109,16 @@ const DateInputV2: React.FC<Props> = ({
...
+        if (!max || datePickerHeaderDate.getFullYear() < max.getFullYear()) {
+          setDatePickerHeaderDate((prev) =>
+            dayjs(prev).add(1, "year").toDate(),
+          );
+        }
         break;
       case "year":
-        setDatePickerHeaderDate((prev) => dayjs(prev).add(1, "year").toDate());
-        setYear((prev) => dayjs(prev).add(10, "year").toDate());
+        if (!max || year.getFullYear() + 10 >= max.getFullYear()) {
```
Comment (Jacobjeevan) [inline src/components/Common/DateInputV2.tsx]: Shouldn't it be less than or equal to here?

**Review by Jacobjeevan — CHANGES_REQUESTED** [review]

File: src/components/Common/DateInputV2.tsx [hunk 2]
```diff
@@ -209,32 +215,72 @@
...
+          datePickerHeaderDate.getMonth(),
+          datePickerHeaderDate.getDate(),
+        ),
+      );
+      setType("date");
+    } else {
+      Notification.Error({
+        msg: outOfLimitsErrorMessage ?? "Cannot select year out of range",
+      });
+    }
```
[the remark this answers is not in the document]
Comment (Jacobjeevan) [reply src/components/Common/DateInputV2.tsx:283]: and this @Rishith25

**Review by Jacobjeevan — CHANGES_REQUESTED** [review]

File: src/components/Common/DateInputV2.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (Jacobjeevan) [reply src/components/Common/DateInputV2.tsx]: And this @Rishith25

**Comment by Rishith25** [comment]
@Jacobjeevan  I have done the changes and  also suggested changes from coderabbitai

**Comment by Jacobjeevan** [comment]
Something I noticed while testing, month selection to default to the min here (same happened for max as well).

image

**Comment by Rishith25** [comment]
@Jacobjeevan Can you review the changes

**Comment by Jacobjeevan** [comment]
> @Jacobjeevan Can you review the changes

@Rishith25 

Do the same for year as well. Test around with different parameters for min and max and check for weird behavior before requesting for review.

**Comment by Rishith25** [comment]
Hey @Jacobjeevan , I have made changes to the min and max limits. The min and max limits for both year and month are now working fine. I have tested all scenarios, and no unusual behavior has been observed. It is ready for review.

**Comment by nihal467** [comment]
@Jacobjeevan can you check the PR

**Review by Jacobjeevan — APPROVED** [review]

**Comment by nihal467** [comment]
LGTM

**Comment by nihal467** [comment]
LGTM

**Review by rithviknishad — APPROVED** [review]

**Pull request merged**